logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/mls/OneOnOneResolver.kt

```diff
@@ -49,6 +53,11 @@ internal class OneOnOneResolverImpl(
         }
     }
 
+    override suspend fun resolveOneOnOneConversationWithUserId(userId: UserId): Either<CoreFailure, ConversationId> =
+        userRepository.getKnownUser(userId).first()?.let {
```

Comment:
```suggestion
        userRepository.getKnownUser(userId).firstOrNull()?.let {
```

Question here. Can be safer to use `firstOrNull` to ensure we are always going to receive `Either.Left(StorageFailure.DataNotFound)` if failure ? 

If I remember correctly `first` can throw an exception.

Reply:
Ah good to know! I'll change it

Reply:
@yamilmedina @typfel Just to add more context:

`first` will only throw if the flow is closed without ever emitting anything.
In this case, _knowing_ that SqlDelight is underneath, and it will immediately emit the current value (or null if there isn't any), using `first` is as safe as `firstOrNull`.

However, there's no benefit in using `first` here either in this case, as `firstOrNull` is a tiny bit more flexible and we deal with nullability anyway.